Lib/fontgoggles/mac/mainWindow.py

```diff
@@ -975,13 +982,21 @@ def featuresChanged(self, sender):
 
     @objc.python_method
     def varLocationChanged(self, sender):
+        """Axis slider value changed"""
+        _, _, _, allInstances, _, _ = self._gatherSidebarInfo(self.project.fonts)
         self.project.textSettings.varLocation = {k: v for k, v in sender.get().items() if v is not None}
         self.textEntryChangedCallback(self.textEntry, updateCharacterList=False)
+        self.variationsGroup.instances.setItems([i[0] for i in allInstances])
+        # Reset instance popup
+        self.variationsGroup.instances.set(0)
 
     @objc.python_method
     def varInstanceChanged(self, sender):
-        self.project.textSettings.varLocation = {k: v for k, v in sender.get().items() if v is not None}
+        """Instance was selected from popup"""
+        _, _, _, allInstances, _, _ = self._gatherSidebarInfo(self.project.fonts)
```

Comment:
These calls to get the `allInstances` are quite clumsy, but I wanted to keep the generation of the list in one spot to avoid confusion. Maybe inside `_gatherSidebarInfo` the instance part could be refactored into a separate method, and in this callback in `varLocationChanged` that separate method would suffice.

Reply:
I was wondering if all of what `_gatherSidebarInfo` returns shouldn't rather be e.g. an `@property` of `Project`. Then `_updateSidebarItems` could just access the properties as needed, and these callbacks could access the instance list as needed. The getters could cache the iteration over the font list internally, to avoid iterating the font list for each property when called without font list changes between calls to different (or same, for that matter) attributes.